front-end/app/service/user-details.service.ts

```diff
@@ -0,0 +1,172 @@
+import { Injectable } from '@angular/core';
```

Comment:
There's a lot of code here, but it looks like it does the job. It would be good to go through this and implement it in a slightly simpler way.